services/director-v2/src/simcore_service_director_v2/api/routes/computations.py

```diff
@@ -0,0 +1,316 @@
+            )
+        await _abort_pipeline_tasks(
+            project, comp_tasks, computation_tasks, celery_client
+        )
+
+    except ProjectNotFoundError as e:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail=str(e)) from e
+
+
+@router.delete(
+    "/{project_id}",
+    summary="Deletes a computation pipeline",
+    response_model=None,
+    status_code=status.HTTP_204_NO_CONTENT,
+)
+async def delete_pipeline(
+    comp_task_stop: ComputationTaskDelete,
+    project_id: ProjectID,
+    project_repo: ProjectsRepository = Depends(get_repository(ProjectsRepository)),
+    computation_pipelines: CompPipelinesRepository = Depends(
+        get_repository(CompPipelinesRepository)
+    ),
+    computation_tasks: CompTasksRepository = Depends(
+        get_repository(CompTasksRepository)
+    ),
+    celery_client: CeleryClient = Depends(get_celery_client),
+):
+    try:
+        # get the project
+        project: ProjectAtDB = await project_repo.get_project(project_id)
+        # check if current state allow to stop the computation
+        comp_tasks: List[CompTaskAtDB] = await computation_tasks.get_comp_tasks(
+            project_id
+        )
+        pipeline_state = get_pipeline_state_from_task_states(
+            comp_tasks, celery_client.settings.publication_timeout
+        )
+        if is_pipeline_running(pipeline_state):
+            if not comp_task_stop.force:
+                raise HTTPException(
+                    status_code=status.HTTP_403_FORBIDDEN,
+                    detail=f"Projet {project_id} is currently running and cannot be deleted, current state is {pipeline_state}",
+                )
+            # abort the pipeline first
+            await _abort_pipeline_tasks(
+                project, comp_tasks, computation_tasks, celery_client
+            )
+
+            MAX_ABORT_WAITING_TIME = 10
+
+            @retry(
+                stop=stop_after_delay(MAX_ABORT_WAITING_TIME),
+                wait=wait_random(0, 2),
+                retry=retry_if_result(lambda result: result is False),
+                reraise=True,
+                before_sleep=before_sleep_log(log, logging.INFO),
+            )
+            async def check_pipeline_stopped() -> bool:
+                comp_tasks: List[CompTaskAtDB] = await computation_tasks.get_comp_tasks(
+                    project_id
+                )
+                pipeline_state = get_pipeline_state_from_task_states(
+                    comp_tasks,
+                    celery_client.settings.publication_timeout,
+                )
+                return is_pipeline_stopped(pipeline_state)
+
+            # wait for the pipeline to be stopped
+            check_pipeline_stopped()
+
+        # delete the pipeline now
+        await computation_tasks.delete_tasks_from_project(project)
+        await computation_pipelines.delete_pipeline(project_id)
+
+    except ProjectNotFoundError as e:
```

Comment:
I think if you call it several times you mean if they run in kind of async/parallel?
else I think nothing should happen since the deletion runs with a where clause